src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -27,14 +26,45 @@ internal partial class DefaultHubDispatcher<THub> : HubDispatcher<THub> where TH
         private readonly IHubContext<THub> _hubContext;
         private readonly ILogger<HubDispatcher<THub>> _logger;
         private readonly bool _enableDetailedErrors;
+        private readonly Func<HubInvocationContext, ValueTask<HubResult>> _invokeMiddleware;
```

Comment:
Instead of funcs, it might be cleaner to create an AggregateHubFilter implementation that is used to call through each of the methods.